src/transformers/models/character_bert/configuration_character_bert.py

```diff
@@ -0,0 +1,127 @@
+        max_position_embeddings (:obj:`int`, `optional`, defaults to 512):
+            The maximum sequence length that this model might ever be used with.
+            Typically set this to something large just in case (e.g., 512 or 1024 or 2048).
+        type_vocab_size (:obj:`int`, `optional`, defaults to 2):
+            The vocabulary size of the :obj:`token_type_ids` passed when calling :class:`~transformers.CharacterBertModel` or
+            :class:`~transformers.TFCharacterBertModel`.
+        initializer_range (:obj:`float`, `optional`, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        layer_norm_eps (:obj:`float`, `optional`, defaults to 1e-12):
+            The epsilon used by the layer normalization layers.
+        use_cache (:obj:`bool`, `optional`, defaults to :obj:`True`):
+            Whether or not the model should return the last key/values attentions (not used by all models). Only
+            relevant if ``config.is_decoder=True``.
+        Example::
+
+        >>> from transformers import CharacterBertModel, CharacterBertConfig
+
+        >>> # Initializing a CharacterBERT helboukkouri/character-bert style configuration
+        >>> configuration = CharacterBertConfig()
+
+        >>> # Initializing a model from the helboukkouri/character-bert style configuration
+        >>> model = CharacterBertModel(configuration)
+
+        >>> # Accessing the model configuration
+        >>> configuration = model.config
+    """
+    model_type = "character_bert"
+    def __init__(
+        self,
+        vocab_size=262,
+        mlm_vocab_size=100000,
+        hidden_size=768,
+        num_hidden_layers=12,
+        num_attention_heads=12,
+        intermediate_size=3072,
+        hidden_act="gelu",
+        hidden_dropout_prob=0.1,
+        attention_probs_dropout_prob=0.1,
+        max_position_embeddings=512,
+        type_vocab_size=2,
+        initializer_range=0.02,
+        layer_norm_eps=1e-12,
+        is_encoder_decoder=False,
+        use_cache=True,
+        **kwargs
+    ):
+        super().__init__(
+            **kwargs
+        )
+
+        self.vocab_size=vocab_size
+        self.mlm_vocab_size=mlm_vocab_size
+        self.max_position_embeddings = max_position_embeddings
+        self.hidden_size = hidden_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.intermediate_size = intermediate_size
+        self.hidden_act = hidden_act
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_probs_dropout_prob = attention_probs_dropout_prob
+        self.initializer_range = initializer_range
+        self.type_vocab_size = type_vocab_size
+        self.layer_norm_eps = layer_norm_eps
+        self.use_cache = use_cache
+        self.tie_word_embeddings = False  # can't tie word embs in CharacterBERT
```

Comment:
This should probably be passed to the superclass instead of defined here

Reply:
Hi Lysandre, I see that `tie_word_embeddings` is already an attribute of the superclass but `use_cache`, `layer_norm_eps` and `type_vocab` do not seem to be defined in PretrainedConfig. Could you elaborate on why it is preferable to not define these attributes in `CharacterBertConfig` ?